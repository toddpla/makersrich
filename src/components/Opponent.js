import React, { Component } from 'react';
import database from '../firebase/firebase'
import PlayerImg from "../assets/player.png"

class Opponent extends Component {

  state = {
    top: -16,
    left: -16
  }

  componentDidMount() {
    database.ref(`/players/${this.props.opponent.id}`).on('value', (snapshot) => {
      console.log('here');
      this.setState({
        ...snapshot.val()
      })
    })
  }

  render() {
    return (
      <div id="player"
        style={{
          position: 'absolute',
          width: '16px',
          top: this.state.top,
          left: this.state.left,
          height: '16px',
          backgroundPosition: 'center',
          backgroundImage: `url(${PlayerImg})`,
          zIndex: 1
        }}
      >
      </div>
    );
  }
}

export default Opponent;
