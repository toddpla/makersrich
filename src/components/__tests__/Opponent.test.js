import React from 'react'
import { shallow } from 'enzyme'
import Opponent from '../Opponent'

test('renders correctly', () => {
  const wrapper = shallow(<Opponent />)
  expect(wrapper).toMatchSnapshot();
});
