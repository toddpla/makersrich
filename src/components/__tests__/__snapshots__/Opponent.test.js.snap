// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Opponent
    opponent={
      Object {
        "uid": 1,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          id="opponent_pick"
          style={
            Object {
              "backgroundImage": "url(pick.png)",
              "backgroundPosition": "center",
              "height": "16px",
              "left": -8,
              "position": "absolute",
              "rotate": 90,
              "top": -16,
              "width": "16px",
            }
          }
        />,
        <div
          id={1}
          style={
            Object {
              "backgroundImage": "url(opponent.png)",
              "backgroundPosition": "center",
              "height": "16px",
              "left": -16,
              "position": "absolute",
              "top": -16,
              "width": "16px",
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "opponent_pick",
          "style": Object {
            "backgroundImage": "url(pick.png)",
            "backgroundPosition": "center",
            "height": "16px",
            "left": -8,
            "position": "absolute",
            "rotate": 90,
            "top": -16,
            "width": "16px",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": 1,
          "style": Object {
            "backgroundImage": "url(opponent.png)",
            "backgroundPosition": "center",
            "height": "16px",
            "left": -16,
            "position": "absolute",
            "top": -16,
            "width": "16px",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            id="opponent_pick"
            style={
              Object {
                "backgroundImage": "url(pick.png)",
                "backgroundPosition": "center",
                "height": "16px",
                "left": -8,
                "position": "absolute",
                "rotate": 90,
                "top": -16,
                "width": "16px",
              }
            }
          />,
          <div
            id={1}
            style={
              Object {
                "backgroundImage": "url(opponent.png)",
                "backgroundPosition": "center",
                "height": "16px",
                "left": -16,
                "position": "absolute",
                "top": -16,
                "width": "16px",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "opponent_pick",
            "style": Object {
              "backgroundImage": "url(pick.png)",
              "backgroundPosition": "center",
              "height": "16px",
              "left": -8,
              "position": "absolute",
              "rotate": 90,
              "top": -16,
              "width": "16px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": 1,
            "style": Object {
              "backgroundImage": "url(opponent.png)",
              "backgroundPosition": "center",
              "height": "16px",
              "left": -16,
              "position": "absolute",
              "top": -16,
              "width": "16px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
